### PR TITLE
add: plot_linestrength example

### PR DESCRIPTION
**Pull request by gagan-aryan**
### Description

This pull request iadds the example to plot linestrengths as a bar plot in the waverange.

### CODE REVIEW

**Comment by erwanp** [comment]
Super nice ! Does it correctly build the docs locally ?

One issue I can see about adding this in the main Examples : it uses CH4 ; which is very long to download and process. The examples will be re-computed for every change to the `develop` branch, so we want to limit data traffic. What about using CO for the online version ?

**Comment by gagan-aryan** [comment]
@erwanp I have replaced it with CO and yes the docs pass locally.

File: examples/plot_linestrengths.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+================================================
+Scale Linestrengths of carbon-monoxide
+================================================
+
+This example scales the linestrengths of CO to Tgas=300 from Tref = 296 and then plots
+the linestrengths against the wavenumbers. We are using :py:func:`~radis.io.hitemp.fetch_hitemp`
+function to retrieve the dataframe from the HITEMP-CO databank. The calculations done
+for scaling use the  equation that can be found in Rotham's paper (equation A11).
+
+"""
```
Comment (erwanp) [inline examples/plot_linestrengths.py:12]: We can use these examples to give information about what's done in the code (good for future spectroscopy tutorials !)

```suggestion
Similar functions are used directly at the hearth of RADIS's SpectrumFactory in the :py:meth:`~radis.lbl.BaseFactory.calc_linestrength_eq` and :py:meth:`~radis.lbl.BaseFactory.calc_linestrength_noneq` methods
"""
```

I also suggest that you : 
- borrow the equation and link to the references of BaseFactory.calc_linestrength_eq docstrings, and add it here. Will look very nice !
- mention this example in BaseFactory.calc_linestrength_eq docstring, by adding 
```
.. minigallery:: radis.lbl.base.BaseFactory.calc_linestrength_eq
```
- make BaseFactory._calc_linestrength_noneq a public method by renaming it BaseFactory.calc_linestrength_noneq

File: examples/plot_linestrengths.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    Tref = 296
+    df = fetch_hitemp(
+        molecule="CO",
+        databank_name="HITEMP",
+        isotope="1, 2, 3",
+        load_wavenum_min=2000,
+        load_wavenum_max=2250,
+    )
+
+    Tgas = 300
```
Comment (erwanp) [inline examples/plot_linestrengths.py:86]: Scaling from 296 to 300 K doesn't sound like a lot of changes. What about scaling up to, let's say, 450 K ?

**Review by erwanp — CHANGES_REQUESTED** [review]

File: examples/plot_linestrengths.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    Qref_arr = np.empty_like(iso_arr, dtype=np.float64)
+    Qgas_arr = np.empty_like(iso_arr, dtype=np.float64)
+    for iso in iso_arr:
+        if iso in iso_set:
+            Qref, Qgas = _calc_Q(molecule, iso, Tref, Tgas)
+            Qref_arr[iso] = Qref
+            Qgas_arr[iso] = Qgas
+
+    df["Qref"] = Qref_arr.take(df.iso)
+    df["Qgas"] = Qgas_arr.take(df.iso)
```
[the remark this answers is not in the document]
Comment (gagan-aryan) [reply examples/plot_linestrengths.py:61]: So we will be able to merge this once we the new approach in merged to the main code ?

[the remark this answers is not in the document]
Comment (erwanp) [reply examples/plot_linestrengths.py:61]: Yes. We can even add your example before the main code is upgraded;  but the example should use the map(dict) approach not to have to modify it afterwards (also, it's faster as you've shown !)

File: examples/plot_linestrengths.py [hunk 4]
```diff
@@ -49,19 +57,15 @@ def _calc_Q(molecule, iso, T_ref, T_gas):
 
     iso_arr = list(range(max(iso_set) + 1))
 
-    Qref_arr = np.empty_like(iso_arr, dtype=np.float64)
-    Qgas_arr = np.empty_like(iso_arr, dtype=np.float64)
+    Qref_Qgas_ratio = {}
+
     for iso in iso_arr:
         if iso in iso_set:
```
Comment (erwanp) [inline examples/plot_linestrengths.py:63]: @gagan-aryan one more comment : why do you need to create iso_arr ? Can't we just do `for iso in iso_set` ?

**Review by erwanp — APPROVED** [review]

**Review by erwanp — APPROVED** [review]

**Pull request merged**